From: Denis Kenzior <denkenz@gmail.com>
Date: Mon, 2 Dec 2019 10:53:55 -0600
Subject: scan: Fix crash when scan is triggered outside iwd

 #0  0x000055555558ee5d in scan_notify (msg=0x55555560b640, user_data=0x0) at src/scan.c:1706
 #1  0x00007ffff7f2c78c in ?? () from /usr/lib/libell.so.0
 #2  0x00007ffff7f299ec in ?? () from /usr/lib/libell.so.0
 #3  0x00007ffff7f28e4a in l_main_iterate () from /usr/lib/libell.so.0
 #4  0x00007ffff7f28efc in l_main_run () from /usr/lib/libell.so.0
 #5  0x00007ffff7f290b9 in l_main_run_with_signal () from /usr/lib/libell.so.0
 #6  0x00005555555639c4 in main (argc=1, argv=0x7fffffffec18) at src/main.c:497
---
 src/scan.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/scan.c b/src/scan.c
index e8133ff..97b37c6 100644
--- a/src/scan.c
+++ b/src/scan.c
@@ -1703,7 +1703,8 @@ static void scan_notify(struct l_genl_msg *msg, void *user_data)
 		else
 			sc->state = SCAN_STATE_PASSIVE;
 
-		sr->start_time_tsf = start_time_tsf;
+		if (sr)
+			sr->start_time_tsf = start_time_tsf;
 
 		break;
 
